File: sprint02/F.js
class Stack {
    constructor() {
        this.stack = [];
    }

    push(item) {
        this.stack.push(item);
    }

    pop() {
        if (this.isEmpty()) {
            console.log('error')
            return;
        }
        return this.stack.pop();
    }

    size() {
        return this.stack.length;
    }

    isEmpty() {
        return this.size() === 0;
    }

    get_max() {
        if (this.isEmpty()) {
            return 'None';
        }

        const extraStack = new Stack();
        let currentItem = null;
        let maximum = -Infinity;
        while (this.isEmpty() === false) {
            currentItem = this.pop();
            if (currentItem > maximum) {
                maximum = currentItem;
            }
            extraStack.push(currentItem);
        }

        while (extraStack.isEmpty() === false) {
            this.push(extraStack.pop());
        }
        return maximum;
    }
}


const _readline = require('readline');

const _reader = _readline.createInterface({
    input: process.stdin
});

const _inputLines = [];
let _curLine = 0;

_reader.on('line', line => {
    _inputLines.push(line);
});

process.stdin.on('end', solve);


function solve() {
    const length = readInt();
    const stack = new Stack();

    for (let i = 0; i < length; i++) {
        const line = readLine();
        const commands = line.split(' ');
        if (commands[0] === 'push') {
            stack.push(commands[1])
        } else if (commands[0] === 'pop') {
            stack.pop();
        } else if (commands[0] === 'get_max') {
            console.log(stack.get_max());

        }
    }
}

function readInt() {
    const n = Number(_inputLines[_curLine]);
    _curLine++;
    return n;
}

function readLine() {
    const line = _inputLines[_curLine];
    _curLine++;
    return line;
}

File: sprint02/F.py
class Stack:
    def __init__(self):
        self.stack = []

    def push(self, item):
        self.stack.append(item)

    def pop(self):
        if (self.isEmpty()):
            print('error')
            return
        return self.stack.pop()

    def size(self):
        return len(self.stack)

    def isEmpty(self):
        if (self.size() == 0):
            return True
        return False

    def get_max(self):
        if self.isEmpty():
            return 'None'

        return max(self.stack)

    def print_stack(self):
        print(f'{" -> ".join(self.stack)}')



stack = Stack()
n = int(input())

while (n):
    input_str =  input()
    input_str = input_str.split()

    if (input_str[0] == 'push'):
        stack.push(input_str[1])
    elif (input_str[0] == 'pop'):
        stack.pop()
    elif (input_str[0] == 'get_max'):
        print(stack.get_max())
    elif (input_str[0] == 'print_stack'):
        stack.print_stack()
    n -= 1

